UefiCpuPkg/PiSmmCpuDxeSmm/NonMmramMapStandaloneMm.c

```diff
@@ -0,0 +1,226 @@
+/** @file
+
+Copyright (c) 2024, Intel Corporation. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include "PiSmmCpuCommon.h"
+
+/**
+  Check SmmProfile is enabled or not.
+
+  @param[in, out]     BaseAddress     Return BaseAddress of SmmProfileData.
+  @param[in, out]     Size            Return Size of SmmProfileData.
+
+  @return TRUE     SmmProfile is enabled.
+          FALSE    SmmProfile is not enabled.
+
+**/
+BOOLEAN
+IsSmmProfileEnabled (

```

Comment:
How about we only find the HOB for once and cache the result? Then we don't need to traverse the HOB list every time the function is called.